rpcs3/Emu/Cell/Modules/cellGame.cpp

```diff
@@ -674,7 +679,7 @@ error_code cellGameGetParamString(s32 id, vm::ptr<char> buf, u32 bufsize)
 	}
 
 	std::string value = psf::get_string(prm->sfo, key);
-	value.resize(bufsize - 1);
+	value.resize(bufsize);
```

Comment:
Do you know why `bufsize - 1` was initially added instead of `bufsize`?

Reply:
No idea, but it leaked info to the VM (And could theoretically crash), no reason it should act that way.

Reply:
It's necessary to ensure that the written string is null-terminated. And zero bufsize case you have already fixed above.

Reply:
If it was smaller than buf size it would still leak to the VM, as the last byte could be of another allocation - better to just add [bufsize - 1] = 0 in the final buffer, I think. To commit that?

Reply:
Leak what? Resize adds zeros, and there is always a zero at value[size].

Reply:
Say value was (bufsize - 2) bytes, and resized to (bufsize - 1) -- a possible situation.
bufsize bytes will be memcpied from value.data, which is a pointer to the data itself, which is only bufsize-1 bytes, with the last one never initialized (Heap read overflow) - leaking data to the VM.

Reply:
Nope. Read about std::string, it always has a one zero char more than its size, which is allowed to read.